drivers/hmis_external_apis/app/models/hmis_external_apis/tc_hmis/importers/loaders/scan_card_loader.rb

```diff
@@ -0,0 +1,78 @@
+###
+# Copyright 2016 - 2023 Green River Data Analysis, LLC
+#
+# License detail: https://github.com/greenriver/hmis-warehouse/blob/production/LICENSE.md
+###
+
+module HmisExternalApis::TcHmis::Importers::Loaders
+  class ScanCardLoader < BaseLoader
+    def initialize(...)
+      super
+
+      @client_lookup = Hmis::Hud::Client.
+        where(data_source_id: data_source.id).
+        pluck(:personal_id, :id).
+        to_h
+
+      @now = Time.current
+    end
+
+    def filename
+      'ScanCodes.xlsx'
+    end
+
+    # why not define this in BaseLoader?
+    def runnable?
+      super && reader.file_present?(filename)
+    end
+
+    def perform
+      Hmis::ScanCardCode.with_deleted.each(&:really_destroy!) if clobber
```

Comment:
deleting all records is fine, we just shouldn't run this after we launch.

Maybe use delete_all since there's no associations on scan card code?
```suggestion
      Hmis::ScanCardCode.with_deleted.delete_all if clobber
```

Reply:
yep, updated